src/panels/lovelace/editor/view-editor/hui-edit-view.ts

```diff
@@ -75,13 +87,25 @@ export class HuiEditView extends LitElement {
       this._config = {};
       this._badges = [];
       this._cards = [];
+      this._users = [];
+      this._visible = [];
     } else {
-      const { cards, badges, ...viewConfig } = this.lovelace!.config.views[
-        this.viewIndex
-      ];
+      const {
+        cards,
+        badges,
+        visible,
+        ...viewConfig
+      } = this.lovelace!.config.views[this.viewIndex];
       this._config = viewConfig;
-      this._badges = badges ? processEditorEntities(badges) : [];
       this._cards = cards;
+      this._badges = badges ? processEditorEntities(badges) : [];
+      // not sure if this is the best way :/
+      if (this.hass) {
+        this._users = await fetchUsers(this.hass);
+      }
+      this._visible = visible
+        ? processViewVisibility(visible, this._users)
+        : [];
```

Comment:
Move all this to `hui-view-visibility-editor`

Reply:
@bramkragten `fetchUsers` is async. Where should I call it? (I'm almost 100% sure that not in the render and not in the constructor)

Reply:
In `firstUpdated`, but don't make `firstUpdated` async, create a second functions that is async that you call from `firstUpdated`

Reply:
ok, done